samza-yarn/src/main/java/org/apache/samza/job/yarn/YarnContainerRunner.java

```diff
@@ -190,13 +196,24 @@ private void startContainer(Path packagePath,
       throw new SamzaContainerLaunchException("IO Exception when writing credentials to output buffer");
     }
 
+    Map<String, LocalResource> localResourceMap = new HashMap<>();
```

Comment:
Not sure I understand completely. I was proposing to remove this duplication by refactoring the logic that populates the package resource inside `LocalizerResourceMapper`.  
- The logic for parsing configs is now spread across 2 places (one to construct the mapping for `yarn.package.path` and the other to construct this resources map). I'm wondering if the two can be in the same class.